src/Build/Graph/ProjectGraphNode.cs

```diff
@@ -0,0 +1,33 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.Build.Evaluation;
+
+namespace Microsoft.Build.Graph
+{
+    /// <summary>
+    /// Represents the node for a particular project in a project graph.
+    /// </summary>
+    internal class ProjectGraphNode
```

Comment:
I would vote for `ProjectGraphItem` instead of `ProjectGraphNode`...

Reply:
I prefer Node (or Vertex) in keeping with most graph-algo standards I've seen.

Reply:
According to Wikipedia they're usually called "vertices, nodes, or points". I feel like I usually see "node". I usually see "item" or "element" in more of a list collection (gut feel).